bulk/text.py

```diff
@@ -1,45 +1,53 @@
-from tornado.ioloop import IOLoop
-
 import numpy as np
 import pandas as pd
-
-from bokeh.io import curdoc
 from bokeh.layouts import column, row
-from bokeh.models import (Button, ColumnDataSource, DataTable, TableColumn, TextInput)
+from bokeh.models import Button, ColumnDataSource, TextInput, DataTable, TableColumn, ColorBar
 from bokeh.plotting import figure
-from bokeh.models import DataTable, TableColumn
 
-def bulk_text(path):
+from .utils import get_color_mapping, get_datatable_columns
+
 
+def bulk_text(path):
     def bkapp(doc):
         df = pd.read_csv(path)
-        saved_idx = []
-        highlighed_idx = []
+        highlighted_idx = []
 
-        source = ColumnDataSource(data=dict())
-        source_orig = ColumnDataSource(data=df)
+        datatable_columns = get_datatable_columns(df)
+        columns = [
+            TableColumn(field=col, title=col) for col in datatable_columns
+        ]
 
         def update(attr, old, new):
-            global highlighed_idx
+            """Callback used for plot update when lasso selecting"""
+            global highlighted_idx
             subset = df.iloc[new]
-            highlighed_idx = new
+            highlighted_idx = new
             subset = subset.iloc[np.random.permutation(len(subset))]
             source.data = subset
 
         def save():
-            df.iloc[highlighed_idx][['text']].to_csv(text_filename.value)
+            """Callback used to save highlighted data points"""
+            df.iloc[highlighted_idx][datatable_columns].to_csv(text_filename.value)
```

Comment:
Before my assumption was that folks are only interested in storing the text. You seem to want to include more. That can be fine, but why do you assume that folks wouldn't be interested in the `x`, `y` and `color` columns?

Reply:
Well, the `x` and `y` columns are pretty hard to interpret if not plotted. One might be interested in the `color` though. But I agree it is debatable whether to save only the `text` column or also the metadata associated with a data point.
Should we agree on saving all the columns but `x` and `y`?

Reply:
Technically, the tool is supposed to merely make a subset of the data. So that suggests to me that the most "basic" behaviour is just to do a subset and nothing else. That would mean we usually just include all variables. It's very little effort to filter them away later and I might prefer moving that responsibility to the end-user if that means we don't have to make an assumption. 

I do think that it'd be nice to have a `--just-text` flag though. But that's perhaps out of scope for this PR. 

For this PR, would you object to just returning all columns?

Reply:
Agree with you it would be nice to have a flag!
Sure, I'll make the changes to return all columns 😃

Reply:
Done in be48a51